BYOA docs (#12046)
PipedreamHQ/pipedream: docs-v2/pages/connected-accounts/oauth-clients.mdx

import Callout from '@/components/Callout'
import ArcadeEmbed from '@/components/ArcadeEmbed'
import { Steps } from 'nextra/components'

# OAuth Clients
By default, most OAuth apps on Pipedream use Pipedream's official OAuth client. That means when you connect an account, you'll see Pipedream's logo and name in the OAuth authorization screen, along with a set of requested permissions (scopes).

> - Pipedream apps solve for a broad range of use cases, which means the scopes requested by Pipedream's official OAuth client may be more than what you need

Reviewer: ```suggestion
```

> - To define the exact scope of access you'd like to grant, you can configure a custom OAuth client for most OAuth apps in Pipedream

Reviewer: ```suggestion
If the scopes are too broad or too narrow for your needs, you can set up your own custom OAuth Client in Pipedream.

When you set up a custom OAuth client for an integrated app in Pipedream, it will replace the Pipedream official OAuth client for that app. This gives you the flexibility to define exactly which scopes should be requested, as well as the opportunity to theme the OAuth client to your brand.
```

## Configuring Custom OAuth Clients
<ArcadeEmbed
    src="https://demo.arcade.software/dSzqDl32VI7hDEPM41Kh?embed"
    title="Create Custom OAuth Client in Pipedream">
</ArcadeEmbed>

<Steps>
### Navigate to the OAuth Clients Page
Open the [OAuth Clients page in your Pipedream account](https://pipedream.com/@/accounts/oauth-clients) and click "New OAuth Client".

### Select the app
Find the app you need. If you can't find what you're looking for, feel free to [submit an integration request](https://pipedream.com/support).

### Enter the required fields
- **Name:** Give the OAuth client a name so it's easy to identify
- **Description:** Optionally add a brief description for additional context
- **Client ID and Secret**: Paste these values from the app's settings that you're configuring (the client secret is sensitive – we'll encrypt and hide it from the UI)
- **Redirect URI:** Copy this Redirect URI and paste it into the app's settings
- **Scopes:** We'll list the scopes from the official Pipedream OAuth client by default. Add or remove scopes as needed, based on your use case.

> Click "Save"

Reviewer: ```suggestion
Finally, click *Save* to create the custom OAuth client.
```
